src/SecureRoute.tsx

```diff
@@ -11,30 +11,48 @@
  */
 
 import * as React from 'react';
-import { useOktaAuth, OnAuthRequiredFunction } from './OktaContext';
+import { useOktaAuth, OnAuthRequiredFunction, OnAuthExpiredFunction } from './OktaContext';
 import { Route, useRouteMatch, RouteProps } from 'react-router-dom';
 import { toRelativeUrl } from '@okta/okta-auth-js';
 
 const SecureRoute: React.FC<{
   onAuthRequired?: OnAuthRequiredFunction;
+  onAuthExpired?: OnAuthExpiredFunction;
 } & RouteProps & React.HTMLAttributes<HTMLDivElement>> = ({ 
-  onAuthRequired, 
+  onAuthRequired,
+  onAuthExpired,
   ...routeProps 
 }) => { 
-  const { oktaAuth, authState, _onAuthRequired } = useOktaAuth();
+  const { oktaAuth, authState, _onAuthRequired, _onAuthExpired } = useOktaAuth();
   const match = useRouteMatch(routeProps);
   const pendingLogin = React.useRef(false);
+  const hasAuthenticated = React.useRef(false);
 
   React.useEffect(() => {
     const handleLogin = async () => {
       if (pendingLogin.current) {
         return;
       }
 
-      pendingLogin.current = true;
-
+      // Save the current route before any redirection
       const originalUri = toRelativeUrl(window.location.href, window.location.origin);
       oktaAuth.setOriginalUri(originalUri);
+
+      // We have previously signed in
+      const hasExpired = hasAuthenticated.current === true;
```

Comment:
Looks like we are able to detect if it's triggered by  initial load or future authState change, how about feeding an extra state object to the `onAuthRequiredFn` callback, then devs can code based on that? 

```
onAuthRequired(oktaAuth, options /* probably not the best name */) 
```

In this way, we make the workaround possible with a minor change. Thoughts?